### ISSUE TITLE
[feat] optimize read, pushdown `limit` to file level for `to_arrow` 

### ISSUE THREAD

**Issue by kevinjqliu** [apache/iceberg-python#1038]
### Feature Request / Improvement

As of now, `limit` is checked only after an entire parquet file is read. 
https://github.com/apache/iceberg-python/blob/d8b5c17cadbc99e53d08ade6109283ee73f0d83e/pyiceberg/io/pyarrow.py#L1360-L1390

Optimization to pushdown limit to the parquet reading level

For more details, see [this comment](https://github.com/apache/iceberg-python/issues/1032#issuecomment-2282819711)

**Comment by soumya-ghosh** [comment]
@kevinjqliu I would like to work on this one.

**Comment by kevinjqliu** [comment]
sure @soumya-ghosh, assigned to you 

The solution might look similar to what is already done for `project_batches` in #1042
https://github.com/apache/iceberg-python/blob/f05b1aedee8451d981188adf68be5e8b360a9ca1/pyiceberg/io/pyarrow.py#L1457-L1479

**Comment by kevinjqliu** [comment]
Closed by #1043 (see [comment](https://github.com/apache/iceberg-python/pull/1043/files?diff=unified&w=0#r1716131334))
